SonarQube.TeamBuild.PreProcessor/Interfaces/IAnalyzerProvider.cs

```diff
@@ -19,6 +20,6 @@ public interface IAnalyzerProvider
         /// </summary>
         /// <param name="projectKey">Identifier for the project being analyzed</param>
         /// <returns>The settings required to configure the build for Roslyn analysis</returns>
-        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey);
+        AnalyzerSettings SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey, string projectBranch);
```

Comment:
We read the project key and the branch individually. We combine them into a projectID and use it to make REST API calls, form URLs and write properties file - in fact I couldn't find methods that take in a project key and not take in a branch. 

So perhaps the AnalysisConfig should store a ProjectId string property that we do not serialize?